### ISSUE TITLE
Crash selecting some images from the device photo library

### ISSUE THREAD

**Issue by pmusolino** [woocommerce/woocommerce-ios#2337]
While I was recording a video while I was reporting another issue on Product Images, I encountered a crash while I was selecting some images from the device photo library.

The attached gif shows the whole process and the crash.
<img src="https://user-images.githubusercontent.com/495617/82670549-a4cc1e00-9c3d-11ea-88e7-bd8236c5f610.gif" width=300 />

## Environment
- Device: iPhone 11 Pro (simulator)
- iOS version: 13.4
- WooCommerce iOS version: 4.3

**Comment by pmusolino** [comment]
Following the same steps (also on different products) and making sure that the alert to give the authorization to access the photo library appears, I can reproduce the crash systematically.
<img width="1101" alt="Schermata 2020-05-22 alle 15 15 14" src="https://user-images.githubusercontent.com/495617/82671386-0d67ca80-9c3f-11ea-819b-8bd474f02cbe.png">

```
Terminating app due to uncaught exception 'NSGenericException', reason: 'Task created in a session that has been invalidated'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23e3dcce __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff50b3b9b2 objc_exception_throw + 48
	2   CFNetwork                           0x00007fff22ffdb92 cfnTranslateCFError + 23986
	3   CFNetwork                           0x00007fff230038ad cfnTranslateCFError + 47821
	4   CFNetwork                           0x00007fff230039dc cfnTranslateCFError + 48124
	5   libdispatch.dylib                   0x0000000112644e8e _dispatch_client_callout + 8
	6   libdispatch.dylib                   0x0000000112653ae2 _dispatch_lane_barrier_sync_invoke_and_complete + 132
	7   CFNetwork                           0x00007fff22ffd086 cfnTranslateCFError + 21158
	8   CFNetwork                           0x00007fff22ffd706 cfnTranslateCFError + 22822
	9   Alamofire                           0x000000010ee34cda $s9Alamofire13UploadRequestC10UploadableO4task7session7adapter5queueSo16NSURLSessionTaskCSo0I0C_AA0C7Adapter_pSgSo012OS_dispatch_H0CtKFSo0ibJ0CyXEfU0_ + 330
	10  Alamofire                           0x000000010ee34b43 $sSo22NSURLSessionUploadTaskCs5Error_pIgozo_ABsAC_pIegrzo_TR + 19
	11  Alamofire                           0x000000010ee34ea4 $sSo22NSURLSessionUploadTaskCs5Error_pIgozo_ABsAC_pIegrzo_TRTA.5 + 20
	12  libswiftDispatch.dylib              0x0000000112247251 $sxs5Error_pIgrzo_xsAA_pIegrzo_lTRTA + 17
	13  libswiftDispatch.dylib              0x000000011223f684 $sSo17OS_dispatch_queueC8DispatchE11_syncHelper33_F417D752D2C4E9330E1C700411CE0C6ALL2fn7execute6rescuexyyyXEXE_xyKXExs5Error_pKXEtKlFyxyKcXEfU_yyXEfU_ + 84
	14  libswiftDispatch.dylib              0x0000000112247071 $sIg_Ieg_TRTA + 17
	15  libswiftDispatch.dylib              0x000000011223a6de $sIeg_IyB_TR + 14
	16  libdispatch.dylib                   0x0000000112644e8e _dispatch_client_callout + 8
	17  libdispatch.dylib                   0x0000000112653ae2 _dispatch_lane_barrier_sync_invoke_and_complete + 132
	18  libswiftDispatch.dylib              0x000000011223fc06 $sSo17OS_dispatch_queueC4sync7executeyyyXE_tFTO + 150
	19  libswiftDispatch.dylib              0x0000000112244390 $sSo17OS_dispatch_queueC4sync7executeyyyXE_tFTOTA + 16
	20  libswiftDispatch.dylib              0x000000011223f55c $sSo17OS_dispatch_queueC8DispatchE11_syncHelper33_F417D752D2C4E9330E1C700411CE0C6ALL2fn7execute6rescuexyyyXEXE_xyKXExs5Error_pKXEtKlF + 252
	21  libswiftDispatch.dylib              0x000000011223fb46 $sSo17OS_dispatch_queueC8DispatchE4sync7executexxyKXE_tKlF + 118
	22  Alamofire                           0x000000010ee33f56 $s9Alamofire13UploadRequestC10UploadableO4task7session7adapter5queueSo16NSURLSessionTaskCSo0I0C_AA0C7Adapter_pSgSo012OS_dispatch_H0CtKF + 1766
	23  Alamofire                           0x000000010ee7d0d0 $s9Alamofire14SessionManagerC6upload33_83E574FCF6E88D5939C6CCAC18E76758LLyAA13UploadRequestCAG10UploadableOF + 384
	24  Alamofire                           0x000000010ee78a17 $s9Alamofire14SessionManagerC6upload_4withAA13UploadRequestC10Foundation3URLV_AA21URLRequestConvertible_ptF + 807
	25  Alamofire                           0x000000010ee7b8b5 $s9Alamofire14SessionManagerC6upload17multipartFormData14usingThreshold4with5queue18encodingCompletionyyAA09MultipartfG0Cc_s6UInt64VAA21URLRequestConvertible_pSo012OS_dispatch_K0CSgyAC0nfG14EncodingResultOcSgtFyycfU_ + 4821
	26  Alamofire                           0x000000010ee7dfb7 $s9Alamofire14SessionManagerC6upload17multipartFormData14usingThreshold4with5queue18encodingCompletionyyAA09MultipartfG0Cc_s6UInt64VAA21URLRequestConvertible_pSo012OS_dispatch_K0CSgyAC0nfG14EncodingResultOcSgtFyycfU_TA + 55
	27  Alamofire                           0x000000010ee08540 $sIeg_IeyB_TR + 48
	28  libdispatch.dylib                   0x0000000112643f11 _dispatch_call_block_and_release + 12
	29  libdispatch.dylib                   0x0000000112644e8e _dispatch_client_callout + 8
	30  libdispatch.dylib                   0x000000011265656d _dispatch_root_queue_drain + 819
	31  libdispatch.dylib                   0x0000000112656ca6 _dispatch_worker_thread2 + 135
	32  libsystem_pthread.dylib             0x00007fff51b9e9f7 _pthread_wqthread + 220
	33  libsystem_pthread.dylib             0x00007fff51b9db77 start_wqthread + 15
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by jaclync** [comment]
thanks a bunch for catching this crash before the release @pmusolino! 

I was able to reproduce on simulators and a test device ([crash log](https://sentry.io/organizations/a8c/issues/1687992588/?project=1458804)). From the stack trace and crash reason (`Task created in a session that has been invalidated`), it sounds like the background session was somehow invalidated. After some testing, I was able to reproduce the crash if I logged in to the app in the same app lifetime. The crash didn't occur to me if the app was already logged in on app launch. @pmusolino could you recall if you logged in to the app in the same app session when you encountered the crash?

**Comment by pmusolino** [comment]
@jaclync as you said, I was able to replicate it deleting the app from the simulator and reinstalling it, so every time I need to do the login, and accept the access to the iOS library. 👍

**Comment by jaclync** [comment]
closing for now with the fix targeting release 4.3 in https://github.com/woocommerce/woocommerce-ios/pull/2342
